include/pmacc/particles/ParticlesBase.kernel

```diff
@@ -229,6 +229,16 @@ struct KernelFillGapsLastFrame
                 uint32_t const
             )
             {
+                if( counterParticles == 0 )
+                {
+                    getPreviousFrameAndRemoveLastFrame(lastFrame, pb, superCellIdx);
```

Comment:
I guess this should be `lastFrame = getPreviousFrameAndRemoveLastFrame(lastFrame, pb, superCellIdx);` ?
Otherwise you have no reference to the `lastFrame` and check for a removed frame in the next line

Reply:
yes absolutly